std/assembly/array.ts

```diff
@@ -167,9 +167,9 @@ export class Array<T> {
     var buffer = this.buffer_;
     var capacity = buffer.byteLength >>> alignof<T>();
     var newLength = length + 1; // safe only if length is checked
-    if (<u32>length >= <u32>capacity) {
+    if (<u32>newLength >= <u32>capacity) {
```

Comment:
If `length` isn't validated explicitly, checking `newLength` might use an overflown value. Once if `length` is known to be valid, `newLength` is known to be in range.

Reply:
so it was legit before?

Reply:
Don't know, what actual issue are you trying to fix?

Reply:
It works in both cases, but modified version more clear in my opinion. And produce more optimal [output](https://github.com/AssemblyScript/assemblyscript/pull/300/files/84d396c4116ac43eb3bd744926da5159aa2b69f1#diff-d48956ee38e0493bf00054580270b5d3R3070)